pkg/secureframe/api.go

```diff
@@ -0,0 +1,114 @@
+package secureframe
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/Jeffail/gabs/v2"
+)
+
+const (
+	restEndpoint = "https://api.secureframe.com"
+)
+
+// makeRequest performs an authenticated HTTP request to the specified endpoint and returns a JSON-friendly byte slice.
+func makeRequest(ctx context.Context, url, method, accessKey, secretKey string) ([]byte, error) {
+	req, err := http.NewRequestWithContext(ctx, method, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Authorization", fmt.Sprintf("%s %s", accessKey, secretKey))
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		return nil, err
+	}
+
+	rb, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return rb, nil
+}
+
+// GetUsers returns a map of noncompliant users and related information.
+func GetUsers(ctx context.Context, accessKey, secretKey, types string) (map[string]map[string]string, error) {
```

Comment:
Avoid the prefix "Get":

https://google.github.io/styleguide/go/best-practices.html#naming-conventions

The genericization of this rule for go is: if you can name a function a Verb, do so. If you can't, give it a Noun. Give it both when necessary for clarity.

Reply:
Updated in [`43b421d` (#4)](https://github.com/chainguard-dev/secureframe-policy-minder/pull/4/commits/43b421d147a39786d38ea0b32986443daa885a12).